### DISCUSSION TITLE
base_kws_vars option for mbcn_train

### DISCUSSION THREAD

**Discussion by sylvainmarchi** [Ouranosinc/xclim#1936] [Questions]
### Setup Information

Xclim version: 0.52.0
Python version: 3.10.14
Operating System: Linux

### Context

I wonder if it would make sense to add an option in sdba.MBCn..train similar to sdba.MBCn.adjust for adapting the frequency of dry days (jitter_under_thresh_value and adapt_freq_thresh).

**Comment by coxipi** [comment]
Sorry Sylvain for responding so late. It's not the first discussion post like this that passes away from my radar. I will try to see if I can configure notification emails like issues, that's mainly how I keep track of Github matters.

Anyhow. In `MBCn.train`, only the multivariate part of the algorithm. In this case, all variables are standardized (da -> (da - mean)/std ), and all training steps are with additive adjustment factor. Not only the 0 value becomes unlikely after normalization for precipitations, but also the adjustment factor take the form:
factor = ref_q - hist_q
and not 
factor = ref_q / hist_q

The possibility of having division by 0 with hist_q from e.g. precipitations would be the main issue. But that's not an issue with normalization and the adjustment factor being formulated as a difference and not a ratio. 

That argument is mainly for jitter_under_thresh. But I believe that adapt_freq_thresh is also useless in this situation, since all the 0 values of precips are shifted due to standardization and the adjustment, again additive rather multiplicative. It is the multiplicative nature of adjustment factor which makes adapt_freq_thresh necessary for regular adjustment of precips.

Maybe the structure of MBCn.train became clearer through our discussions in recent months and you will agree with me. But if there's something I've overlooked, please let me know!

Cheers, 
Éric
